src/main/java/de/tum/in/www1/artemis/domain/Exercise.java

```diff
@@ -445,13 +445,14 @@ else if (participation.getExercise() instanceof ModelingExercise || participatio
      * @param participation the participation whose results we are considering
      * @return the latest relevant result in the given participation, or null, if none exist
      */
-    public Result findLatestRatedResultWithCompletionDate(Participation participation) {
+    public Result findLatestRatedResultWithCompletionDate(Participation participation, ZonedDateTime assessmentDueDate) {
```

Comment:
I think it we should have a boolean parameter (`ignoreAssessmentDueDate`).

It seems like a workaround that `filterForCourseDashboard` has to pass in a value that is generated by the same class (`findLatestRatedResultWithCompletionDate` could also just call `getAssessmentDueDate` or even use the class property without the getter). 
Also the meaning of the method invocation in `ResultResource.getResultsForExercise` with a null value for the assessmentDueDate is not self explanatory (is there no assessmentDueDate? should it be ignored?).
This could also lead to somebody just passing in any date they like even though this is part of the domain of the `Exercise`.

Reply:
i really like your suggestion

Reply:
i added the parameter